mmv1/third_party/terraform/utils/compute_instance_helpers.go.erb

```diff
@@ -384,34 +381,38 @@ func flattenEnableDisplay(displayDevice *computeBeta.DisplayDevice) interface{}
 
 // Terraform doesn't correctly calculate changes on schema.Set, so we do it manually
 // https://github.com/hashicorp/terraform-plugin-sdk/issues/98
-func schedulingHasChange(d *schema.ResourceData) bool {
+// 1st bool is a change that requires a reboot and the second bool is a change that doesn't
+func schedulingHasChange(d *schema.ResourceData) (bool, bool) {
 	if !d.HasChange("scheduling") {
 		// This doesn't work correctly, which is why this method exists
 		// But it is here for posterity
-		return false
+		return false, false
 	}
 	o, n := d.GetChange("scheduling")
 	oScheduling := o.([]interface{})[0].(map[string]interface{})
 	newScheduling := n.([]interface{})[0].(map[string]interface{})
 	originalNa := oScheduling["node_affinities"].(*schema.Set)
 	newNa := newScheduling["node_affinities"].(*schema.Set)
+	if !reflect.DeepEqual(newNa, originalNa) {
```

Comment:
@upodroid I believe this condition is returning false positives. Like in `TestAccComputeInstance_attachedDiskUpdate`, when the config is being updated from `testAccComputeInstance_attachedDisk` to `testAccComputeInstance_addAttachedDisk`

Reply:
Just so we don't spend duplicate time, I plan on resolving this and pushing it over the finish line from here. Thanks @upodroid !

Reply:
I agree. I'm stuck